### PR TITLE
Color palette settings button hover tootlip fix.

### LINKED ISSUE #14685: Color settings: selected color doesn't display its tooltip on hover
In the color settings, when a color is selected, its tooltip is not displayed on hover:

Screenshot 2019-03-28 at 15 33 17

It does appear on focus (when tabbing with the keyboard).

This is not only an accessibility issue (the name of the control is not exposed in any way) but also an user experience issue. Users may want to know a color name even when it's selected.

### PR DESCRIPTION
**Pull request by v9ai**
## Description
Fixes https://github.com/WordPress/gutenberg/issues/14685
Adding `z-index` fixes the issue.

## How has this been tested?
Has been tested locally, more details in gif from below

## Screenshots
Peek 2019-03-28 23-12

## Types of changes
Had added few styles to color palette settings button in active state.

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by nicolad** [comment]
@afercia can you please confirm that this PR is fixing the issue: https://github.com/WordPress/gutenberg/issues/14685 ?

**Comment by afercia** [comment]
@nicolad yes, it fixes it. Thanks! I don't have the powers to approve PRs though.

**Comment by gziolo** [comment]
> Thanks! I don't have the powers to approve PRs though.

Everyone can approve :)

File: packages/components/src/color-palette/style.scss
Comment (gziolo) [inline packages/components/src/color-palette/style.scss]: @jasmussen - should there be special handling for `z-index` used?

**Review by gziolo — APPROVED** [review]
It looks like a valid fix. There is one remaining code maintainability related question to be answered before merging.

File: packages/components/src/color-palette/style.scss
Comment (jasmussen) [reply packages/components/src/color-palette/style.scss]: My impression is that's always good. But I'm honestly not confident, sometimes a z index of 1 is just to elevate a child in a context that isn't subject to the mess it can otherwise create. I know @aduth has thoughts.

Comment (gziolo) [reply packages/components/src/color-palette/style.scss]: I'll leave it to him then :)

**Comment by afercia** [comment]
Note: if using `z-index` is not desirable, there's an alternative approach to fix this. Setting `pointer-events: none` on the selected button SVG should do the trick.

**Comment by gziolo** [comment]
> Note: if using `z-index` is not desirable, there's an alternative approach to fix this. Setting `pointer-events: none` on the selected button SVG should do the trick.

I guess `z-index` is perfectly fine, I just wanted to double check whether we need to use `z-index()` helper.

**Comment by jasmussen** [comment]
Yep, z-index is fine. And if this PR is semi urgent or can make the cut by being merged now, feel free to and if Andrew objects to the lack of the z-index helper later on, I volunteer to fix that after the fact.

**Comment by gziolo** [comment]
@nicolad - thanks for the fix. Let's wait for feedback and take action only if necessary.

**Comment by nicolad** [comment]
> @nicolad - thanks for the fix. Let's wait for feedback and take action only if necessary.

Sure, I can add a follow-up PR based on feedback, if it will be the case.

File: packages/components/src/color-palette/style.scss
Comment (aduth) [reply packages/components/src/color-palette/style.scss]: My impression has been to always use `_z-index` function for any `z-index`, with the mapping serving as a centralized authority to serve in proactively avoiding aggressively and endlessly increasing a `z-index` value to "combat" others.

https://github.com/WordPress/gutenberg/blob/master/assets/stylesheets/_z-index.scss

But:

1. For a case like this, where the value is `1`, perhaps it's reasonable to make exceptions
2. In any case, we don't codify this guideline one way or the other. Whatever we decide, it ought to be written down, e.g. `coding-guidelines.md` document

**Pull request merged**
